README.md

```diff
@@ -58,3 +60,70 @@ If the slice size is increased then folding occurs:
 
 Probably you can find a parameter set that works for a large group of patients. Maybe one parameter set works for all, but maybe you need to have a few different presets (small, medium, large)
+
+## Characterize Transform Matrix
+
+It is often difficult to understand what a transform matrix is doing just by inspection.  All the information is in those 12 numbers, but not in an easily understood format. CharacterizeTransformMatrix is a simple utility module which tries to quickly give you any information you might want to know about what a transformation matrix is doing.  For example, is it a rigid transformation or is there scaling?  If there is scaling, what are the scale factors and stretch directions?  Is there rotation?  If so, what is the axis of rotation and how much rotation occurs around that axis?  Alternatively, if we break down the rotation into a sequence of rotations around coordinate axes, what is the rotation about each axis?
+
+### To use
+Open the module and select the transform node you want to know about.  An analysis such as the following will appear in the text box below:
+
+```
+Scale factors and stretch directions (eigenvalues and eigenvectors of stretch matrix K):
+  f0: +0.012% change in direction [1.00, 0.03, -0.08]
+  f1: -2.843% change in direction [-0.08, -0.10, -0.99]
+  f2: +3.248% change in direction [0.04, -0.99, 0.10]
+This transform is not rigid! Total volume changes by +0.325%, and maximal change in one direction is +3.248%
+The rotation matrix portion of this transformation rotates 15.0 degrees ccw (if you look in the direction the vector points) around a vector which points to [0.76, -0.59, -0.27] (RAS)
+Broken down into a series of rotations around axes, the rotation matrix portion of the transformation rotates 
+  11.8 degrees ccw around the positive R axis, then 
+  8.4 degrees cw around the positive A axis, then 
+  5.0 degrees cw around the positive S axis
+Lastly, this transformation translates, shifting:
+  +194.2 mm in the R direction
+  +73.4 mm in the A direction
+  -1170.3 mm in the S direction
+```
+This analysis is for the matrix 
+```
+0.985821 0.0570188 -0.157817 194.155 
+-0.0873217 1.01 -0.192319 73.4412 
+0.14329 0.203373 0.94 -1170.25 
+0 0 0 1 
+```
+### Some Decomposition Details
+This module uses polar decomposition to describe the components of a 4x4 transform matrix. The decomposition has the form: `H = T * R * K`, where `H` is the full homogeneous transformation matrix (with 0,0,0,1 as the bottom row), `T` is a translation-only matrix, `R` is a rotation-only matrix, and `K` is a stretch matrix. `K` can further be decompsed into three scale matrices, which can each be characterized by a stretch direction (an eigenvector) and a stretch factor (the associated eigenvalue). Points to be transformed are on the right, so the order of operations is stretching first, then rotation, then translation. 
```

Comment:
Can you justify the choice of polar decomposition? Why don't we use `vtkMath::Orthogonalize3x3`?

Reply:
The clearest example I was following (I think from https://math.stackexchange.com/questions/237369/given-this-transformation-matrix-how-do-i-decompose-it-into-translation-rotati) used polar decomposition and that approach seemed appropriate.  It may be that an approach based on `vtkMath::Orthogonalize3x3` would also work, and I don't have a strong justification for why one approach might be better than the other.